### PR TITLE
Introduce LLM-based single-table model.

### PR DESCRIPTION
**Pull request by MooooCat**
## Description

For a long time, LLM has been used to understand and generate various types of data. In fact, LLM also has certain capabilities in tabular data generation. More over, it has some abilities that cannot be achieved by traditional (based on GAN methods or statistical methods) . 

In this PR, we introduce `sdgx.models.LLM.single_table.SingleTableGPT.SingleTableGPTModel`, our first synthetic data generation model  integrating LLM.

## Motivation and Context

Compared with existing models, `SingleTableGPTModel` implements two new features:
- Generation without Data: No training data is required, synthetic data can be generated based on metadata data;
- Off-Table Feature Inference: Infer new column data based on the existing data in the table and the knowledge mastered by LLM. 

In addition, `SingleTableGPTModel` can directly generate data without complicated and time-consuming steps such as manual labeling and feature engineering, which will save a lot of operator time and allow them to focus on creative work.

## How has this been tested?

We currently provide some test cases at tests/models/test_singletableGPT.py. This test file contains some content returned by GPT. We will not repeatedly request GPT in the unit test to avoid consuming a large amount of tokens.

I will continue to improve these test cases.

## Types of changes

- [ ] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by MooooCat** [comment]
Some of the comments are still incomplete at the moment, I will add them as soon as possible.

File: sdgx/models/LLM/single_table/SingleTableGPT.py
Comment (Wh1isper) [inline sdgx/models/LLM/single_table/SingleTableGPT.py:1]: We should use snakecase for the filename

File: tests/models/test_singletableGPT.py
Comment (Wh1isper) [inline tests/models/test_singletableGPT.py:1]: We should use snakecase for the filename, maybe `test_singletable_gpt.py`?

Comment (MooooCat) [reply tests/models/test_singletableGPT.py:1]: This makes sense, I'll change the filename.

File: sdgx/models/LLM/base.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    This is a potential way to generate data that cannot be made public and contains sensitive information.
+    """
+
+    _metadata = None
+    """
+    the metadata.
+    """
+
+    off_table_features = []
```
Comment (Z712023) [inline sdgx/models/LLM/base.py:39]: The introduction of variable `off_table_features` is an interesting idea. :)

**Review by Z712023 — APPROVED** [review]

**Comment by Wh1isper** [comment]
How about release 0.2.0 for it?

**Comment by MooooCat** [comment]
> How about release 0.2.0 for it?

Good idea, I need make another few changes (google colab examples, readme updates...) before releasing.

**Pull request merged**